### PR TITLE
resource/aws_msk_cluster: Update encryption_info.encryption_in_transit.client_broker default to match API default

### LINKED ISSUE #10673: aws_msk_cluster : default value for client_broker : "TLS"
Hi 

I am creating MSK cluster and if i do not keep any  encryption_in_transit. It picks up "TLS" value for client_broker and cluster is created. But if I do terrform apply (without making any changes.). It forces below change : 
      encryption_info.0.encryption_in_transit.0.client_broker: "TLS" => "TLS_PLAINTEXT" (forces new resource)

Moreover if I provide value as : 
    encryption_in_transit {
    client_broker = "TLS"
    in_cluster  = true
  }

Cluster creation fails with : 
Error: Error refreshing state: 1 error occurred:
	* aws_msk_cluster.atropos_prod: 1 error occurred:
	* aws_msk_cluster.atropos_prod: aws_msk_cluster.XXXX: failed requesting bootstrap broker info for "arn:aws:kafka:ap-south-1:286817589435:cluster/XXXX/684bd0d9-eecf-454d-b209-306ead63dda6-4" : BadRequestException: You can't get bootstrap broker nodes for a cluster in FAILED state.
	status code: 400, request id: 81a984a9-2646-46e7-86af-236c0a88b329

Terrform version : Terraform v0.11.14

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #10673

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* resource/aws_msk_cluster: Update `encryption_info` `encryption_in_transit` `client_broker` argument default to match API default
```

Output from acceptance testing:

```
--- PASS: TestAccAWSMskCluster_basic (1595.73s)
--- PASS: TestAccAWSMskCluster_BrokerNodeGroupInfo_EbsVolumeSize (1760.04s)
--- PASS: TestAccAWSMskCluster_EncryptionInfo_EncryptionAtRestKmsKeyArn (1603.40s)
--- PASS: TestAccAWSMskCluster_EncryptionInfo_EncryptionInTransit_ClientBroker (1603.85s)
--- PASS: TestAccAWSMskCluster_EncryptionInfo_EncryptionInTransit_InCluster (1598.02s)
--- PASS: TestAccAWSMskCluster_EnhancedMonitoring (1603.10s)
--- PASS: TestAccAWSMskCluster_LoggingInfo (1806.72s)
--- PASS: TestAccAWSMskCluster_NumberOfBrokerNodes (2036.98s)
--- PASS: TestAccAWSMskCluster_OpenMonitoring (1722.68s)
--- PASS: TestAccAWSMskCluster_Tags (1605.63s)

--- PASS: TestAccAWSMskClusterDataSource_Name (1608.30s)
```

### CODE REVIEW

File: aws/data_source_aws_msk_cluster_test.go [hunk 1]
```diff
@@ -22,8 +23,8 @@ func TestAccAWSMskClusterDataSource_Name(t *testing.T) {
 				Config: testAccMskClusterDataSourceConfigName(rName),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttrPair(resourceName, "arn", dataSourceName, "arn"),
-					resource.TestCheckResourceAttrSet(dataSourceName, "bootstrap_brokers"),
-					resource.TestCheckResourceAttrSet(dataSourceName, "bootstrap_brokers_tls"),
+					resource.TestCheckResourceAttr(resourceName, "bootstrap_brokers", ""),
+					resource.TestMatchResourceAttr(resourceName, "bootstrap_brokers_tls", regexp.MustCompile(`^(([-\w]+\.){1,}[\w]+:\d+,){2,}([-\w]+\.){1,}[\w]+:\d+$`)), // Ordering not guaranteed
```
Comment (anGie44) [inline aws/data_source_aws_msk_cluster_test.go:27]: could we use `TestCheckResourceAttrPair` checks in exchange of these checks on the resource?

Comment (bflad) [reply aws/data_source_aws_msk_cluster_test.go:27]: Ah sorry the Go comment wasn't clear, unfortunately the ordering of the hostnames within the string response isn't guaranteed between read calls, so the resource and the data source may have differing responses. This resource attribute may be a candidate in the future for us going against the API and making a `TypeSet` for better usability.

**Comment by bflad** [comment]
Rebased the documentation conflicts and I'll clarify that comment in the testing.

File: aws/data_source_aws_msk_cluster_test.go (hunk 1 above)
Comment (anGie44) [reply aws/data_source_aws_msk_cluster_test.go]: Ohhh gotcha, makes sense!

**Pull request merged**